### ISSUE TITLE
xnio-file-watcher exception

### ISSUE THREAD

**Issue by tdiesler** [wildfly-extras/wildfly-camel#345]
```
12:15:37,413 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-7) JBAS017535: Unregistered web context: /camel-lucene-tests
12:15:37,418 ERROR [stderr] (xnio-file-watcher[Watcher for /Users/tdiesler/git/wildfly-camel/itests/standalone/target/wildfly-8.2.0.Final/standalone/tmp/vfs/temp/tempba30efa92c23d7f8/content-b18aaf1b4173a8fb/]-0) Exception in thread "xnio-file-watcher[Watcher for /Users/tdiesler/git/wildfly-camel/itests/standalone/target/wildfly-8.2.0.Final/standalone/tmp/vfs/temp/tempba30efa92c23d7f8/content-b18aaf1b4173a8fb/]-0" java.nio.file.ClosedWatchServiceException
12:15:37,418 ERROR [stderr] (xnio-file-watcher[Watcher for /Users/tdiesler/git/wildfly-camel/itests/standalone/target/wildfly-8.2.0.Final/standalone/tmp/vfs/temp/tempba30efa92c23d7f8/content-b18aaf1b4173a8fb/]-0) 	at sun.nio.fs.AbstractWatchService.checkOpen(AbstractWatchService.java:80)
12:15:37,418 ERROR [stderr] (xnio-file-watcher[Watcher for /Users/tdiesler/git/wildfly-camel/itests/standalone/target/wildfly-8.2.0.Final/standalone/tmp/vfs/temp/tempba30efa92c23d7f8/content-b18aaf1b4173a8fb/]-0) 	at sun.nio.fs.AbstractWatchService.checkKey(AbstractWatchService.java:92)
12:15:37,418 ERROR [stderr] (xnio-file-watcher[Watcher for /Users/tdiesler/git/wildfly-camel/itests/standalone/target/wildfly-8.2.0.Final/standalone/tmp/vfs/temp/tempba30efa92c23d7f8/content-b18aaf1b4173a8fb/]-0) 	at sun.nio.fs.AbstractWatchService.take(AbstractWatchService.java:119)
12:15:37,419 ERROR [stderr] (xnio-file-watcher[Watcher for /Users/tdiesler/git/wildfly-camel/itests/standalone/target/wildfly-8.2.0.Final/standalone/tmp/vfs/temp/tempba30efa92c23d7f8/content-b18aaf1b4173a8fb/]-0) 	at org.xnio.nio.WatchServiceFileSystemWatcher.run(WatchServiceFileSystemWatcher.java:85)
12:15:37,419 ERROR [stderr] (xnio-file-watcher[Watcher for /Users/tdiesler/git/wildfly-camel/itests/standalone/target/wildfly-8.2.0.Final/standalone/tmp/vfs/temp/tempba30efa92c23d7f8/content-b18aaf1b4173a8fb/]-0) 	at java.lang.Thread.run(Thread.java:745)
```

**Comment by tdiesler** [comment]
The WatchService is closed by
```
Thread [MSC service thread 1-5] (Suspended (breakpoint at line 208 in WatchServiceFileSystemWatcher))	
	owns: FileResourceManager  (id=432)	
	owns: UndertowDeploymentInfoService  (id=433)	
	WatchServiceFileSystemWatcher.close() line: 208	
	FileResourceManager.close() line: 194	
	ServletResourceManager.close() line: 90	
	CachingResourceManager.close() line: 166	
	IoUtils.safeClose(Closeable) line: 134	
	UndertowDeploymentInfoService.stop(StopContext) line: 402	
	ServiceControllerImpl$StopTask.stopService(Service<? extends S>, StopContext) line: 2056	
	ServiceControllerImpl$StopTask.run() line: 2017	
	ServiceContainerImpl$ContainerExecutor(ThreadPoolExecutor).runWorker(ThreadPoolExecutor$Worker) line: 1142	
	ThreadPoolExecutor$Worker.run() line: 617	
	ServiceContainerImpl$ServiceThread(Thread).run() line: 745	
```

**Comment by tdiesler** [comment]
Blocked by: https://issues.jboss.org/browse/XNIO-240

Waiting for [3.3.1.Final](https://github.com/xnio/xnio/tags)

**Comment by tdiesler** [comment]
Fixed in wildfly-9.0.0.CR2
